### ISSUE TITLE
Ability to abort multi_curl "Mid-queue"

### ISSUE THREAD

**Issue by pritcham** [php-curl-class/php-curl-class#706]
Many APIs return data in pages (e.g. 50 results per page), with different methods of telling you there are no more pages/data left (some return a blank resultset, others with a message telling you there's no more data to pull etc).

The $multi_curl->success callback can be used to check for such messages, but there doesn't appear to be a way of telling it to stop processing any more requests that may be in the queue.

Is this something that could be implemented?

Thanks in advance
Martin

**Comment by zachborboa** [comment]
Hi. Yes, stopping additional requests in the queue should be possible.

Possible implementation:
```php
$multi_curl = new MultiCurl();
$multi_curl->error(function ($instance) use ($multi_curl) {
    $multi_curl->stop();
});

$multi_curl->addGet('https://httpbin.org/status/503');
$multi_curl->addGet('https://httpbin.org/status/503');
$multi_curl->addGet('https://httpbin.org/status/503');

$multi_curl->start();
```

**Comment by mohammadraquib** [comment]
Hello @zachborboa,

Can you please explain how can I implement this "stop" function in the MultiCurl queue.

Thanks

**Comment by zachborboa** [comment]
Implemented in #708. Thanks
